### ISSUE TITLE
`resource.WithoutBuiltin()` is ineffective if used by itself within `resource.New()`

### ISSUE THREAD

**Issue by RichieSams** [open-telemetry/opentelemetry-go#1813]
### Description

`resource.WithoutBuiltin()` is ineffective if used by itself within `resource.New()`
`resource.New()` will return nil, which downstream code treats as "no config, give me the defaults".

For example, `prometheus.InstallNewPipeline()`, will in turn call `controller.New()`. If the resource passed to `controller.New()` is nil, then `controller.New()` will assume that you want the defaults. AKA, all the things we wanted to remove with `resource.WithoutBuiltin()`

### Environment

- OS: Linux (but shouldn't matter)
- Architecture: x64 (but shouldn't matter)
- Go Version: 1.16
- opentelemetry-go version: v0.19.0

### Steps To Reproduce

Example code:
```golang
res, err := resource.New(
	context.Background(),
	resource.WithoutBuiltin(),
)
if err != nil {
	return nil, fmt.Errorf("Failed to initialize Prometheus exporter - %w", err)
}

// Create the metrics route
metricsHandler, err := prometheus.InstallNewPipeline(prometheus.Config{}, controller.WithResource(res))
if err != nil {
	return nil, fmt.Errorf("Failed to initialize Prometheus exporter - %w", err)
}
```

All prometheus metrics created will have all the built-in labels added. (Service, sdk, etc.)

### Workaround

```golang
type NullDetector struct{}

// Detect implements Detector
func (NullDetector) Detect(ctx context.Context) (*resource.Resource, error) {
	return nil, nil
}
```

```golang
res, err := resource.New(
	context.Background(),
	resource.WithoutBuiltin(),
	resource.WithDetectors(&telemetry.NullDetector{}),
)
if err != nil {
	return nil, fmt.Errorf("Failed to initialize Prometheus exporter - %w", err)
}

// Create the metrics route
metricsHandler, err := prometheus.InstallNewPipeline(prometheus.Config{}, controller.WithResource(res))
if err != nil {
	return nil, fmt.Errorf("Failed to initialize Prometheus exporter - %w", err)
}
```

### Expected behavior

Passing `resource.WithoutBuiltin()` to `resource.New()`, should result in no built-in labels being added.

### Possible solution

`resource.WithoutBuiltin()` should both nil out all the built-in detectors, but also add a NullDetector, so that we're guaranteed to have a single "valid" detector on return.

**Comment by Aneurysm9** [comment]
I think this might be addressed by #1810, can you take a look and see if that would meet your needs?

**Comment by RichieSams** [comment]
Oh great. That's exactly the solve.
